<?xml version="1.0"?>
<launch>
  <arg name="gemweightspath" value="/home/yujc/robotws/DSLAM_one/src/ROS-DSLAM/script/ROS_GEM/weights/gem.pth" />
  <arg name="datasetpath" value="/home/yujc/robotws/dataset/image_503_loop/" />
  
  
  <node name="publish" pkg="dslam_sp" required="true" type="publish_merge" args="-p $(arg datasetpath)"/>
  
  <node name="sp_ORB" pkg="dslam_sp" required="true" type="orb_EF" output="screen">
    <remap from="image" to="/merge/img_depth_file" />
  </node>

  
  <node name="vo" pkg="dslam_sp" required="true" type="visual_odometry_orb" output="screen">
    <remap from="feature" to="/orb_EF/featurepoints_descriptors" />
  </node>

    
  <node name="gen_tf" pkg="dslam_sp" required="true" type="generate_track_node.py" output="screen">
    <remap from="relpose" to="/visual_odometry/transform_relative" />
  </node>

  <node name="gem_extract" pkg="dslam_sp" required="true" type="extract_feature_ros.py"  args="-w $(arg gemweightspath)" output="screen" >
    <remap from="PRrepresentor" to="GEM_PRrepresentor" />
  </node>

  <node name="gem_receiver" pkg="dslam_sp" required="true" type="receiver_ros.py" output="screen">
    <remap from="PRrepresentor" to="GEM_PRrepresentor" />
    <!-- <remap from="looppose" to="GEM_LOOPS" /> -->
  </node>

  <node name="fnames2pose" pkg="dslam_sp" required="true" type="vo_loop_fnames_orb"  args="-p $(arg datasetpath)" output="screen">
    <remap from="filenames" to="/loopfnames" />
    <remap from="VO_loop_relpose" to="/looppose" />
  </node>


</launch>
